[AlainS87] iP

AlainS87 opened the pull request:
# **Chatbot Alain**

# Alain frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* FAST SUPER FAST to use
* allows input with txt files and/or with mannual key-in

---

#All you need to do is:
1. Item 1
2. Item 2
3. Item 3

# Features:
- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)

---
If you Java programmer, you can use it to practice Java too. Here's the main method:
<pre>
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
</pre>

:smile:

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

AlainS87 committed:
Level0 completed

AlainS87 committed:
level1 completed

AlainS87 committed:
level1 rectified

AlainS87 committed:
level3 completed

AlainS87 committed:
level2 completed

AlainS87 committed:
level3 completed

AlainS87 committed:
level 4 completed

AlainS87 committed:
level 5 completed

AlainS87 committed:
level 6 completed

AlainS87 committed:
fix the problem in level 6 after using test_ui

AlainS87 committed:
modify test-ui tetscase

AlainS87 committed:
level 7 completed

AlainS87 committed:
Merge branch 'branch-Level-7'

AlainS87 committed:
level-8 minimal finished

AlainS87 committed:
Merge branch 'branch-Level-8'

AlainS87 committed:
more OOP applications

AlainS87 committed:
package created for alain

AlainS87 committed:
Merge branch 'add-gradle-support'

AlainS87 committed:
gradle setup done

AlainS87 committed:
Testcases created b Junit passed

AlainS87 committed:
alain.jar created in build/lib

AlainS87 committed:
JavaDoc is done for all classes

AlainS87 committed:
A-CodingStandard is acomplished

AlainS87 committed:
find function completed

AlainS87 committed:
Merge branch 'branch-Level-9'

AlainS87 committed:
style check further modified

AlainS87 committed:
GUI finished

AlainS87 committed:
Merge tag 'GUI'

GSgiansen commented on src/main/java/Chatbot/alain/Deadlines.java:
```diff
@@ -0,0 +1,33 @@
+/**
+ * Represents a task with a specific deadline.
+ */
+public class Deadlines extends Task {
```
Since this class refers to a singular Deadline Object, I feel that the name should be changed to Deadline rather than Deadlines to indicate singularity

GSgiansen commented on src/main/java/Chatbot/alain/Events.java:
```diff
@@ -0,0 +1,42 @@
+/**
+ * Represents a task that is an event occurring within a specific time range.
+ */
+public class Events extends Task {
```
Similar to Deadlines, singular task should be in its Singular term, thus the naming would be better off as "Event" class

GSgiansen commented on src/main/java/Chatbot/alain/ToDos.java:
```diff
@@ -0,0 +1,27 @@
+/**
+ * Represents a task of type "To-Do" without a specific deadline or time.
+ */
+public class ToDos extends Task {
```
Similarly, ToDos would be better if replaced with ToDo

GSgiansen commented on src/main/java/Chatbot/alain/Ui.java:
```diff
@@ -0,0 +1,139 @@
+ * User interface class for displaying messages and interactions with the Alain chatbot.
+ */
+public class Ui {
+    private String logo = "    _     __         _     _____   ___   __\n"
```
Good work on the use of correct access modifiers!

GSgiansen commented on src/main/java/Chatbot/alain/Storage.java:
```diff
@@ -0,0 +1,212 @@
+        FileWriter writer = new FileWriter(listFile, false);
+        String filecontent = "";
+        if (except) {
+            filecontent += "Oops! Seems like there is an exception detected in your input\n";
```
filecontent should be replaced as fileContent for variable!

GSgiansen commented on src/main/java/Chatbot/alain/GUI_Ui.java:
```diff
@@ -0,0 +1,149 @@
+        output += this.showlines();
+        output += " Noted. I've removed this task:\n"
+                + "   " + removedTask + "\n"
+                + " Now you have " + list.size() + " tasks in the list.\n";
```
indentation of the line should be 8 spaces! so possibly adding 8 more spaces from the second + operator

GSgiansen reviewed (commented):
Overall Good job on the coding quality! Aside from some minor changes to the class names and variable names, the rest LGTM! 👍

adhigop13 commented on src/main/java/Chatbot/alain/Storage.java:
```diff
@@ -0,0 +1,212 @@
+     * @return The transformed time string.
+     * @throws AlainException If an exception occurs during the transformation.
+     */
+    public static String stringToTimeString(String inputTime) throws AlainException {
```
Can this be handled in another class eg Parse.java for better encapsulation?

adhigop13 commented on src/main/java/Chatbot/alain/Storage.java:
```diff
@@ -0,0 +1,212 @@
+                    }
+                    continue;
+                }
+                if (isToDo) {
```
Perhaps the use of if, else if, else (or) switch-case structure would have been more suitable?

adhigop13 commented on src/main/java/Chatbot/alain/TaskList.java:
Good, clear naming helps the reader understand code easily! 👍

adhigop13 commented on src/main/java/Chatbot/alain/Storage.java:
```diff
@@ -0,0 +1,212 @@
+public class Storage {
+    private String filePath;
+    private Ui ui = new Ui();
+    private Boolean alrBye;
```
Perhaps a variable name that is more reflective of our Boolean naming standards could be used? eg isEnd

adhigop13 commented on src/main/java/Chatbot/alain/Storage.java:
```diff
@@ -0,0 +1,212 @@
+            filecontent += "Oops! Seems like there is an exception detected in your input\n";
+            filecontent += msg + "\n";
+        } else {
+            filecontent += "____________________________________________________________\n"
```
Perhaps, this could have been encapsulated into the Ui class instead of handling Ui elements here?

adhigop13 commented on src/main/java/Chatbot/alain/Ui.java:
```diff
@@ -0,0 +1,139 @@
+        this.showlines();
+        String output = " Nice! I've marked this task as not done yet:\n"
+                + "   " + list.getTask(Integer.parseInt(numericPart) - 1) + "\n";
+        System.out.print(output);
```
Can these print statements be handled in the Ui class instead? 🤔

AlainS87 committed:
Add Assertions.java and assertion methods to verify task operations

- Implement Assertions class with methods to validate:
  - Addition of new To-do, Deadline, and Event tasks.
  - Proper removal of a task from TaskList.
- Useful for debugging and ensuring task operations behave as expected.

AlainS87 committed:
Refactor and enhance codebase for better quality

- Improve code readability and structure
- Optimize functions for efficiency

AlainS87 committed:
Merge pull request #1 from AlainS87/branch-A-Assertions

Add Assertions.java and assertion methods to verify task operations

AlainS87 committed:
Merge pull request #2 from AlainS87/branch-A-CodeQuality

Branch a code quality

AlainS87 committed:
feat: Add chronological sorting for deadlines

- Implemented a function to sort deadline items based on their dates.
- Allows users to view their deadlines in a chronological order.

AlainS87 committed:
Merge tag 'BCD-Extension'

AlainS87 committed:
more personal elements added, GUI is also improved

AlainS87 committed:
A-UserGuide finished

AlainS87 committed:
change the main class to Launcher

AlainS87 committed:
more exception handled

AlainS87 committed:
more type of errors specified

AlainS87 committed:
save function modified

AlainS87 committed:
no message

AlainS87 committed:
save function modified

AlainS87 committed:
Update README.md

AlainS87 committed:
Update README.md

AlainS87 committed:
Update README.md
